activerecord/test/cases/timestamp_test.rb

```diff
@@ -457,6 +464,14 @@ def test_index_is_created_for_both_timestamps
   ensure
     ActiveRecord::Base.connection.drop_table(:foos)
   end
+
+  def test_datetime_precision_applied_to_timestamp
+    d = Parrot.create!(name: 'Bogdan')
+    in_instance = d.created_at
+    from_db = Parrot.find(d.id).created_at
+    assert_equal from_db.usec, in_instance.usec
+    assert_equal from_db, in_instance
```

Comment:
I find this test hard to follow, and I don't think this is relevant to the behavior of AR::Timestamp. Can we move this to a separate test in something like `test/cases/type/date_time_test.rb`, and have it look something like this?

```ruby
model = Model.new(date_time_attr_that_isnt_auto_touched: Time.now)
assert_equal model.date_time_attr, model.tap(&:save).reload.date_time_attr
```